receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -191,17 +193,75 @@ func (prw *prometheusRemoteWriteReceiver) translateV2(_ context.Context, req *wr
 			intraRequestCache[hashedLabels] = rm
 		}
 
+		scopeName, scopeVersion := prw.extractScopeInfo(ls)
+		metricName := ls.Get(labels.MetricName)
+		unit := ""
+		// TODO: Like UnitRef, we should assign the HelpRef to the metric.
+		if ts.Metadata.UnitRef > 0 && ts.Metadata.UnitRef < uint32(len(req.Symbols)) {
+			unit = req.Symbols[ts.Metadata.UnitRef]
+		}
```

Comment:
```suggestion
		// TODO: Like UnitRef, we should assign the HelpRef to the metric.
		if ts.Metadata.UnitRef < 0 && ts.Metadata.UnitRef > uint32(len(req.Symbols)) {
			return erro
		}
		unit = req.Symbols[ts.Metadata.UnitRef]
```

Reply:
Could you also investigate if the package that manages the symbols table has something built-in to validate this?

Reply:
And also add a test to make sure we keep error-ing in the future